tests/end_to_end/Solmate/test_erc20.py

```diff
@@ -38,142 +34,160 @@ async def test_should_set_name_symbol_and_decimals(self, erc_20):
-        async def test_should_transfer_from(self, erc_20, owner, others):
+        @pytest.mark.parametrize(
+            "initial_allowance, final_allowance", ((TEST_SUPPLY, 0), (MAX_INT, MAX_INT))
+        )
+        async def test_should_transfer_and_update_allowance(
+            self, erc_20, owner, others, initial_allowance, final_allowance
+        ):
             from_wallet = others[0]
             to_wallet = others[1]
 
             await erc_20.mint(
                 from_wallet.address,
                 TEST_SUPPLY,
-                caller_address=owner,
+                caller_eoa=owner,
             )
 
+            balance_sender_before = await erc_20.balanceOf(from_wallet.address)
+            balance_receiver_before = await erc_20.balanceOf(to_wallet.address)
+
             await erc_20.approve(
                 owner.address,
-                TEST_SUPPLY,
-                caller_address=from_wallet,
+                initial_allowance,
+                caller_eoa=from_wallet,
             )
             assert await erc_20.transferFrom(
                 from_wallet.address,
                 to_wallet.address,
                 TEST_SUPPLY,
-                caller_address=owner,
+                caller_eoa=owner,
             )
-            assert await erc_20.totalSupply() == TEST_SUPPLY
-            assert await erc_20.allowance(from_wallet.address, owner.address) == 0
-            assert await erc_20.balanceOf(from_wallet.address) == 0
-            assert await erc_20.balanceOf(to_wallet.address) == TEST_SUPPLY
 
-        async def test_should_transfer_from_with_infinite_approve(
-            self, erc_20, owner, others
-        ):
-            from_wallet = others[0]
-            to_wallet = others[1]
+            balance_sender_after = await erc_20.balanceOf(from_wallet.address)
+            balance_receiver_after = await erc_20.balanceOf(to_wallet.address)
 
-            await erc_20.mint(
-                from_wallet.address,
-                TEST_SUPPLY,
-                caller_address=owner,
+            assert (
+                await erc_20.allowance(from_wallet.address, owner.address)
+                == final_allowance
             )
+            assert balance_sender_before - balance_sender_after == TEST_SUPPLY
+            assert balance_receiver_after - balance_receiver_before == TEST_SUPPLY
 
-            await erc_20.approve(owner.address, MAX_INT, caller_address=from_wallet)
-            assert await erc_20.transferFrom(
-                from_wallet.address,
-                to_wallet.address,
-                TEST_SUPPLY,
-                caller_address=owner,
-            )
-            assert await erc_20.totalSupply() == TEST_SUPPLY
-            assert await erc_20.allowance(from_wallet.address, owner.address) == MAX_INT
-            assert await erc_20.balanceOf(from_wallet.address) == 0
-            assert await erc_20.balanceOf(others[1].address) == TEST_SUPPLY
-
+        @pytest.mark.xfail(
```

Comment:
did we create an issue to track the dojo issue?

Reply:
not really, I think that we should create an issue to track all `xfail` tests, like every week.
(Note: I've replaced some `mark.skip` by `mark.xfail` after re-reading the pytest doc; `xfail` and `xpass` are indeed to be used to keep the test running, not breaking the CI, but indicate that the intented behavior doesn't work)

Reply:
yeah maybe we should find a way for automatically check all the xfail and see if there are resolved or how we can resolve them

Reply:
a weekly telegram bot